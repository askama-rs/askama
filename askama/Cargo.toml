[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
