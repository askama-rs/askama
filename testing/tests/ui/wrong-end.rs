use askama::Template;

#[derive(Template)]
#[template(source = "{% for _ in 1..=10 %}{% end %}", ext = "txt")]
struct For;

#[derive(Template)]
#[template(source = "{% macro test() %}{% end %}", ext = "txt")]
struct Macro;

#[derive(Template)]
#[template(source = "{% filter upper %}{% end %}", ext = "txt")]
struct Filter;

#[derive(Template)]
#[template(source = "{% match () %}{% when () %}{% end %}", ext = "txt")]
struct Match;

#[derive(Template)]
#[template(source = "{% block body %}{% end %}", ext = "txt")]
struct Block;

#[derive(Template)]
#[template(source = "{% if true %}{% end %}", ext = "txt")]
struct If;

#[derive(Template)]
#[template(source = "{% if true %}{% endfor %}", ext = "txt")]
struct IfFor;

fn main() {}
