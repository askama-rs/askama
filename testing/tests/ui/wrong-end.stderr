error: expected `endfor` to terminate `for` node, found `end`
 --> <source attribute>:1:24
       "end %}"
 --> tests/ui/wrong-end.rs:4:21
  |
4 | #[template(source = "{% for _ in 1..=10 %}{% end %}", ext = "txt")]
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endmacro` to terminate `macro` node, found `end`
 --> <source attribute>:1:21
       "end %}"
 --> tests/ui/wrong-end.rs:8:21
  |
8 | #[template(source = "{% macro test() %}{% end %}", ext = "txt")]
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endfilter` to terminate `filter` node, found `end`
 --> <source attribute>:1:21
       "end %}"
  --> tests/ui/wrong-end.rs:12:21
   |
12 | #[template(source = "{% filter upper %}{% end %}", ext = "txt")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endmatch` to terminate `match` node, found `end`
 --> <source attribute>:1:30
       "end %}"
  --> tests/ui/wrong-end.rs:16:21
   |
16 | #[template(source = "{% match () %}{% when () %}{% end %}", ext = "txt")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endblock` to terminate `block` node, found `end`
 --> <source attribute>:1:19
       "end %}"
  --> tests/ui/wrong-end.rs:20:21
   |
20 | #[template(source = "{% block body %}{% end %}", ext = "txt")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endif` to terminate `if` node, found `end`
 --> <source attribute>:1:16
       "end %}"
  --> tests/ui/wrong-end.rs:24:21
   |
24 | #[template(source = "{% if true %}{% end %}", ext = "txt")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `endif` to terminate `if` node, found `endfor`
 --> <source attribute>:1:16
       "endfor %}"
  --> tests/ui/wrong-end.rs:28:21
   |
28 | #[template(source = "{% if true %}{% endfor %}", ext = "txt")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
